File: 601-700/0692-Top-K-Frequent/0692.java
class Solution {
  public List<String> topKFrequent(String[] words, int k) {
    Map<String, Integer> cnt = new HashMap<String, Integer>();
    for (String word : words) {
      cnt.put(word, cnt.getOrDefault(word, 0) + 1);
    }
    List<String> rec = new ArrayList<String>();
    for (Map.Entry<String, Integer> entry : cnt.entrySet()) {
      rec.add(entry.getKey());
    }
    Collections.sort(rec, new Comparator<String>() {
      public int compare(String word1, String word2) {
        return cnt.get(word1) == cnt.get(word2) ? word1.compareTo(word2) : cnt.get(word2) - cnt.get(word1);
      }
    });
    return rec.subList(0, k);
  }
}

File: 601-700/0692-Top-K-Frequent/0692.py
class Solution:
  def topKFrequent(self, words: List[str], k: int) -> List[str]:
    m = {}
    for word in words:
      if word not in m:
        m[word] = 1
      else:
        m[word] += 1
    sorted_list = sorted(m.items(), key=lambda item: (-item[1], item[0]))
    return [item[0] for item in sorted_list][:k]